### PR TITLE
Added separate permission to trigger DAG

### PR DESCRIPTION
**Pull request by AVMusorin**
To divide editing DAG and triggering it. Ability to grant permission per a dag can make RBAC more flexible. But it's also possible to set up `can trigger on Dags`.

Also, there is a [discussion](https://github.com/apache/airflow/discussions/30121) with related topic .

### CODE REVIEW

**Comment by uranusjr** [comment]
I think we need a db migration to populate trigger permissions from edit permissions to roles/users? Otherwise this would cause things to break when an existing Airflow installation is upgraded.

**Comment by AVMusorin** [comment]
> I think we need a db migration to populate trigger permissions from edit permissions to roles/users? Otherwise this would cause things to break when an existing Airflow installation is upgraded.

AFAIU we don't need to create migration for roles because of they are syncing at start (managed by [AIRFLOW__WEBSERVER__UPDATE_FAB_PERMS](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#update-fab-perms))

**Comment by potiuk** [comment]
> AFAIU we don't need to create migration for roles because of they are syncing at start (managed by [AIRFLOW__WEBSERVER__UPDATE_FAB_PERMS](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#update-fab-perms))

Does it include new roles that people create and assign them EDIT_DAG permissions ? 

Are you sure of that ? (I am not - but checking if you checked this migration scenario). It's a heavily breaking change if it does not work. And I would be surprised if it did to be honest. How would the underlying know that we should replace EDIT_DAG with EDIT_DAG/TRIGGER_DAG combo ? Those seem to be two unrelated permissions now?

**Comment by potiuk** [comment]
Note - this is the case when users created their own roles - not relying on built-in USER/ADMIN roles, and assigned them permissions manually - whch is perfectly possible and expected and our user do that).

**Comment by potiuk** [comment]
Also, there is a very similar PR here : https://github.com/apache/airflow/pull/30581  - and I had the same doubts, so much that I thought that it's the same issue we are talking about .

**Comment by AVMusorin** [comment]
> Also, there is a very similar PR here : #30581 - and I had the same doubts, so much that I thought that it's the same issue we are talking about .

Yes, you are right. I'll create migration.

**Comment by AVMusorin** [comment]
@potiuk take a look, please on this PR. 
As we have very similar changes with https://github.com/apache/airflow/pull/30581, I think both of PR's could be added as they have different permissions and together can make permission system better. I can wait for that changes are merged and rebase on them to resolve conflicts if needed.

File: docs/apache-airflow/img/airflow_erd.svg
Comment (uranusjr) [inline docs/apache-airflow/img/airflow_erd.svg:1]: The new image is not rendering!

Comment (AVMusorin) [reply docs/apache-airflow/img/airflow_erd.svg:1]: fixed

File: airflow/migrations/versions/0125_2_6_0_added_separate_permissions_to_trigger_.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+from airflow.security import permissions
+from airflow.www.app import cached_app
+
+# revision identifiers, used by Alembic.
+revision = "bc91cfaccdba"
+down_revision = "98ae134e6fff"
+branch_labels = None
+depends_on = None
+airflow_version = "2.6.0"
```
Comment (eladkal) [inline airflow/migrations/versions/0125_2_6_0_added_separate_permissions_to_trigger_.py]: 2.6.0 is already released so this migration needs to be updated

Comment (AVMusorin) [reply airflow/migrations/versions/0125_2_6_0_added_separate_permissions_to_trigger_.py]: Fixed.

**Comment by potiuk** [comment]
conflicts to solve I am afraid.

**Pull request closed without merging**